docs/src/explanations/Connection-operators-.md

```diff
@@ -0,0 +1,364 @@
+  })
+  io.a <> io.b
+}
+```
+Below we can see the resulting Verilog for this example:
+```scala mdoc
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion: 
+The Verilog remained the same without incurring errors, showing that the `<>` operator is commutative.
+
+
+
+
+## Concept 2: `:=` means assign ALL LHS signals from the RHS, regardless of the direction on the LHS.
+Using the same experiment code as above, we set to test for the function of `:=`
+We replace all instances of `<>` with `:=` in the sample code above.
+(Scastie link to the experiment: https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.stage.ChiselStage
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  // connect producer to I/O
+  p.io.a := io.in
+  // connect producer  to consumer
+  c.io.a := p.io.b
+  // connect consumer to I/O
+  io.out := c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.a := io.b
+}
+```
+Below we can see the resulting error message for this example:
+```scala mdoc:crash
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+The := operator goes field-by-field on the LHS and attempts to connect it to the same-named signal from the RHS. If something on the LHS is actually an Input, or the corresponding signal on the RHS is an Output, you will get an error as shown above.
+
+## Concept 3: Always Use `:=` to assign DontCare to Wires
+When assigning `DontCare` to something that is not directioned, should you use `:=` or `<>`? 
+We will find out using the sample codes below:
+( Scastie link for the experiment:https://scastie.scala-lang.org/Shorla/ZIGsWcylRqKJhZCkKWlSIA/1)
+
+```scala mdoc:silent:reset
+import chisel3._
+import chisel3.stage.ChiselStage
+import chisel3.util.DecoupledIO
+
+class Wrapper extends Module{
+  val io = IO(new Bundle {
+  val in = Flipped(DecoupledIO(UInt(8.W)))
+  val out = DecoupledIO(UInt(8.W))
+  })
+  val p = Module(new PipelineStage)
+  val c = Module(new PipelineStage) 
+  //connect Producer to IO
+  io.in := DontCare
+  p.io.a <> DontCare
+  val tmp = Wire(Flipped(DecoupledIO(UInt(8.W))))
+  tmp := DontCare
+  p.io.a <> io.in
+  // connect producer to consumer
+  c.io.a <> p.io.b
+  //connect consumer to IO
+  io.out <> c.io.b
+}
+class PipelineStage extends Module{
+  val io = IO(new Bundle{
+    val a = Flipped(DecoupledIO(UInt(8.W)))
+    val b = DecoupledIO(UInt(8.W))
+  })
+  io.b <> io.a
+}
+```
+Below we can see the resulting error for this example:
+```scala mdoc
+ChiselStage.emitVerilog(new Wrapper)
+```
+### Conclusion:
+When `<>` was used to assign the unidrectioned wire `tmp` to DontCare, we got an error.
```

Comment:
The code above uses `:=` to assign `tmp`. We did not get an error. This text needs to be adjusted ... 
```suggestion
If `<>` were used to assign the unidrectioned wire `tmp` to DontCare, we would get an error. But in the example above, we used `:=` and no errors occurred.
```